src/_guides/openapi/specification/v3.1/the-perfect-modern-openapi-workflow.md

```diff
@@ -0,0 +1,660 @@
+a lot of code review, and the folks helping with design reviews might have to
+learn various languages, multiple frameworks, and understand all the intricacies
+of those frameworks less obvious conventions. Moving the review to OpenAPI
+removes all of that, and helps to democratize and centralize the process and
+have more folks involved, instead of just those who understand the code.
+
+To design reviews easier, Bump.sh can show a list of relevant changes between
+the OpenAPI in the a pull request and the latest deployed document. This helps
+skip staring at infinite lines of YAML/JSON, trying to understand what, if
+anything, meaningfully changed. New properties, changed property validations, or
+breaking changes like a removed endpoint or new required property. 
+
+```yaml
+# .github/workflows/api-diff.yaml
+name: API Changes
+permissions:
+  contents: read
+  pull-requests: write
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  changes:
+    name: Detect and preview API changes
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Comment pull request with API diff
+        uses: bump-sh/github-action@v1
+        with:
+          doc: <BUMP_DOC_ID>
+          token: ${{secrets.BUMP_TOKEN}}
+          file: api/openapi.yaml
+          command: diff
+
+      - name: API Preview pull request with API diff
+        uses: bump-sh/github-action@v1
+        with:
+          doc: <BUMP_DOC_ID>
+          token: ${{secrets.BUMP_TOKEN}}
+          file: api/openapi.yaml
+          command: preview
+```
+
+#### API Linting Again 
+
+Linting was used in the design and development phase, running locally in the CLI
+or editor, but linting is useful in the governance phase too. By running API
+linting on pull requests, much of the design review process can be automated. 
+
+For example, instead of wasting human time spotting naming convention issues, or
+making somebody want to quit because their entire job is reminding API
+developers "this endpoint does not have a description, please add one", the
+robots can do that work instead.
+
+Configuring the linter to return errors and warnings as annotations on the
+problematic lines in a pull request makes it even easier, helping developers
+spot the problems with their OpenAPI as quickly as they would any syntax errors
+in their code.
+
+![](/images/guides/the-perfect-modern-openapi-workflow/vacuum-annotations.png)
+
+Organizations can use this concept to produce their own
+[rulesets](https://quobix.com/vacuum/rulesets/custom-rulesets/), which are
+basically the old idea of an API Style Guide, but instead of writing out a
+manifesto which nobody will ever read, it can be an automated process. By adding
+new rules over time as more problems pop up or best practices evolve, it helps
+to iteratively improve not just the API in question, but all future and existing
+APIs in the organization.
```

Comment:
It seems that the Github Action workflow file example is missing in this paragraph? Don't you want to add an example of how to get the vacuum linter (and annotations) running?